src/app/core/utils/api-token/api-token.service.ts

```diff
@@ -88,16 +108,16 @@ export class ApiTokenService {
           const cookieContent = apiToken?.apiToken ? JSON.stringify(apiToken) : undefined;
           if (cookieContent) {
             cookiesService.put('apiToken', cookieContent, {
-              expires: new Date(Date.now() + 3600000),
-              secure: true,
+              expires: apiToken.options?.expires ?? new Date(Date.now() + ApiTokenService.DEFAULT_EXPIRY_TIME),
```

Comment:
As far as I can see `apiToken` object doesn't contain `options` key. How do you map the `expires_in` and `refresh_expires_in` characteristics of the token resource object?